src/coreclr/interop/trackerobjectmanager.cpp

```diff
@@ -84,7 +84,10 @@ namespace
 
     STDMETHODIMP HostServices::ReleaseDisconnectedReferenceSources()
     {
-        return InteropLibImports::WaitForRuntimeFinalizerForExternal();
+        // This could lead to deadlock if finalizer thread is trying to get back to this thread, because we are
+        // not pumping anymore. Disable this for now. See: https://github.com/dotnet/runtime/issues/109538.
+        // return InteropLibImports::WaitForRuntimeFinalizerForExternal();
```

Comment:
```suggestion
        // We'd like to call InteropLibImports::WaitForRuntimeFinalizerForExternal() here, but
        // this could lead to deadlock if the finalizer thread is trying to get back to this thread, because we are
        // not pumping anymore. Disable this for now. See: https://github.com/dotnet/runtime/issues/109538.

```

Nit: I'd prefer a wording like this that doesn't have the commented-out code.

Reply:
Updated for both CoreCLR and NativeAOT! 🙂